### PR TITLE
Fix regression of new controller runtime in shoot creation of tests

### PR DESCRIPTION
**Pull request by schrodit**
**What this PR does / why we need it**:
Adjust create-shoot test to the new controller-runtime client list function.

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

**Review by rfranzke — DISMISSED** [review]
/lgtm, thanks for spotting and fixing!

File: .test-defs/cmd/helper.go [hunk 1]
```diff
@@ -174,7 +174,7 @@ func UpdateBackupInfrastructureAnnotations(backup *gardenv1beta1.BackupInfrastru
 // GetBackupInfrastructureOfShoot returns the BackupInfrastructure object of the shoot
 func GetBackupInfrastructureOfShoot(ctx  context.Context, shootGardenerTest *framework.ShootGardenerTest, shootObject *gardenv1beta1.Shoot) (*gardenv1beta1.BackupInfrastructure, error) {
 	backups := &gardenv1beta1.BackupInfrastructureList{}
-	err := shootGardenerTest.GardenClient.Client().List(ctx, &client.ListOptions{Namespace: shootObject.Namespace}, backups)
+	err := shootGardenerTest.GardenClient.Client().List(ctx, backups, client.UseListOptions(&client.ListOptions{Namespace: shootObject.Namespace}))
```
Comment (ialidzhikov) [inline .test-defs/cmd/helper.go]:
```suggestion
	err := shootGardenerTest.GardenClient.Client().List(ctx, backups, client.InNamespace(shootObject.Namespace))
```

Comment (schrodit) [reply .test-defs/cmd/helper.go]: Thanks didn't know that helper:)

**Review by ialidzhikov — DISMISSED** [review]
/approve

**Review by rfranzke — APPROVED** [review]

**Pull request merged**